src/ProxyProtocol.h

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (C) 1996-2018 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_PROXYPROTOCOL_H
+#define SQUID_PROXYPROTOCOL_H
+
+#include "base/RefCount.h"
+#include "ip/Address.h"
+#include "sbuf/SBuf.h"
+
+// https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt
+namespace ProxyProtocol {
+namespace Two {
+
+// unused yet
+typedef enum {
+    // protocol registered types
+    PP2_TYPE_UNKNOWN = 0,
+    PP2_TYPE_ALPN = 0x01,
+    PP2_TYPE_AUTHORITY = 0x02,
+    PP2_TYPE_CRC32C = 0x03,
+    PP2_TYPE_NOOP = 0x04,
+    PP2_TYPE_SSL = 0x20,
+    PP2_SUBTYPE_SSL_VERSION = 0x21,
+    PP2_SUBTYPE_SSL_CN = 0x22,
+    PP2_SUBTYPE_SSL_CIPHER = 0x23,
+    PP2_SUBTYPE_SSL_SIG_ALG = 0x24,
+    PP2_SUBTYPE_SSL_KEY_ALG = 0x25,
+    PP2_TYPE_NETNS = 0x30,
+
+    // extra (protocol-unrelated) types used for logging
+    PP2_EXTRA_TYPE_VERSION = 0x101,
```

Comment:
I would add a trailing comment to clarify: `// larger than 255 to avoid clashes with possible TLV type IDs`